### PR TITLE
Add  and remove tasks

### PR DESCRIPTION
**Pull request by Moathal**
# This pull request is mainly for project functionality

## Changes to main JavaScript `index.js`

- Add eventListeners to add task

- Add new function to add new task

- Add evenListener for clicking any of the items inside a task

- Add function for changing the description

## changes to class module `method.js`

- Complete add task function

- Add function for deleting a task

- Add function to rearrange array & fix order after deleting

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,27 +1,71 @@
...
+newDescription.addEventListener('keydown', (e) => {
+  const description = newDescription.value;
+  e.preventDefault();
+  if (description.length > 0 && e.code === 'Enter') {
+    methods.addTask(description, false, methods.tasks.length);
+    methods.resetList(tasksList);
+    localStorage.setItem('tasks', JSON.stringify(methods.tasks));
+    newDescription.value = '';
+  }
+});
```
Comment (Nemwel-Boniface) [inline src/index.js]: - [x] When adding tasks, I realized that the indexes of the tasks in the local storage are not updating correctly. There are two "index: 1" in the local storage which is not the desired behavior.

**This is an example of what I mean:**

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @Moathal           :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

You have done an great job so far with your to-do list application :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [x] For some reason, when I try typing something to the input field for adding a new task, I am not able to see anything being outputted. I am basically unable to see what I am typing. I have only been able to test the application by pasting the text I copied from somewhere else. Kindly look into why this is the case.

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (Moathal) [reply src/index.js]: Thanks a lot, fixed the bugs and waiting for another review

**Review by leonardodiasb — CHANGES_REQUESTED** [review]
# INVALID ❌ ❌ ❌ 

Hi @Moathal, 👋

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌
 - Your linters are not passing. Kindly check and fix all the linters before submitting a Pull Request.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

✅ Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_⚠️ Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

----
_Invalid Code Review Request does not count into the code reviews limit._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 

## Hi @Moathal 👋, 

Well done implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ App working as expected
✔️ Descriptive PR
✔️ No linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**